src/seraphis_mocks/mock_send_receive.cpp

```diff
@@ -216,6 +216,119 @@ void send_sp_coinbase_amounts_to_users(const std::vector<std::vector<rct::xmr_am
     ledger_context_inout.commit_unconfirmed_txs_v1(coinbase_tx);
 }
 //-------------------------------------------------------------------------------------------------------------------
+void construct_tx_for_mock_ledger_v1(const legacy_mock_keys &local_user_legacy_keys,
+    const jamtis::mocks::jamtis_mock_keys &local_user_sp_keys,
+    const InputSelectorV1 &local_user_input_selector,
+    const FeeCalculator &tx_fee_calculator,
+    const rct::xmr_amount fee_per_tx_weight,
+    const std::size_t max_inputs,
+    const std::vector<std::tuple<rct::xmr_amount, jamtis::JamtisDestinationV1, TxExtra>> &outlays,
+    const std::size_t legacy_ring_size,
+    const std::size_t ref_set_decomp_n,
+    const std::size_t ref_set_decomp_m,
+    const SpBinnedReferenceSetConfigV1 &bin_config,
+    const MockLedgerContext &ledger_context,
+    SpTxSquashedV1 &tx_out,
+    std::vector<jamtis::JamtisPaymentProposalSelfSendV1> &selfsend_payment,
+    std::vector<jamtis::JamtisPaymentProposalV1> &normal_payment)
+{
```

Comment:
If I checked correctly, the following happened here: There is a `construct_tx_for_mock_ledger_v1` that is *almost* usable for your purposes, except that it does not give back those two kinds of payment proposals. Your reaction: You take that method, duplicate all 100 lines or so, add two output parameters and modify the duplicated code a bit to set those parameters.

Well, of course that *technically* works, but the resulting code duplication is a software engineering no-go. People looking at this in the future will never be fully sure that the two parameters are the only difference, because it's just too hard to compare the code of the two methods. And if something changes in transaction construction, you will have to modify forever *two* methods, and I guarantee you it does not take long until somebody modifies something and does not notice that there are two methods. Result: One broken and faulty transaction contruction method.

There are basically two sane and reasonable ways to do what is needed here:

A) You keep a single method, add the two parameters you need, and modify the places where the method is called in existing Seraphis library code accordingly.

B) You take the existing method, add the two parameters you need. Then you write a quite short and simple *new* method with the old signature that calls your modified method and just "throws away" the two vectors. This way you don't have to modify any code that calls the existing method.

Seeing that there are only few calls to the method, I would probably go way A).